run/config.txt
!!python/object/new:config.ConfigClass
dictitems:
  advanced:
    _d storage: ../data/
    atmospheric storage: ../data/
    construction grid _d: !!python/object/apply:numpy.core.multiarray._reconstruct
      args:
      - &id001 !!python/name:numpy.ndarray ''
      - !!python/tuple
        - 0
      - !!binary |
        Yg==
      state: !!python/tuple
      - 1
      - !!python/tuple
        - 600
      - &id002 !!python/object/apply:numpy.dtype
        args:
        - f8
        - false
        - true
        state: !!python/tuple
        - 3
        - <
        - null
        - null
        - null
        - -1
        - -1
        - 0
      - false
      - !!binary |
        gnZJaMIlPD3jfDgShnc/PTBK5+2/lkE9B6WUj8CpQz1beGz+X/tFPczJZ7cAk0g9bYrBKOR4Sz0M
        tob8RLZOPegIcD66KlE9je/Xt/0wUz3e0xNpX3RVPZaxqHQU/Fc9jmsplSvQWj0QF0/GpvldPREe
        YPpLwWA9YkDxhyC7Yj2lHuzxm/BkPe5j6xbHaGc9Zpc8NX8raj0DPWD2jkFtPaWYUTdlWnA9z9H9
        NBdIcj1M+Ou0AXB0PaV6p2EC2XY9l9YiLcaKeT3EbGfC4Y18Pcj6Otns6389cr9gYdDXgT1wzDJI
        ffKDPctP5aCwTIY9AJCvOejtiD05nL4KhN6LPTrwBuLgJ489BuwjGjtqkT2KlBq5+3eTPe5F96W8
        w5U97ii4rM1UmD0zP1RWWzObPdXmv/CIaJ49kgFp1Eb/oD1bmVuJagCjPeYuR8QRPqU9q+KAaV+/
        pz3OOazOTYyqPQB/4CLIra09fifqauOWsD1rxUGst4uyPQVXN86bu7Q926w/4YYttz30Xfk7Qum5
        PYXgSEeC97w9NQ+KGwExwD3OHPOD0RnCPRa6FhJHPMQ9VWW0Dy6fxj1Q3U4BIErJPRE//dmbRcw9
        pfHlCSGbzz1r8MbepqrRPf3tJ1cAwNM9XAHWdz8U1j3zG+kYz67YPZSB/v/5l9s9/aaISAXZ3j04
        Z630Jj7hPWZPutq0RuM9RCGUIKaM5T3AVY0QOBfoPTNsPIOC7uo9tUGdpZEb7j1t96ZkQdTwPZEA
        VU1S0PI9I4+skU0I9T06iP8FRIP3PSOwoM4bSfo99oS6h6li/T0cb0sy5mwAPsxM88/GXAI+vy+U
        0CGHBD65GY6j3PIGPopIMuqspwk+HrobBTGuDD4VLGDDBQgQPlYFUvEA7BE+ru1yXQ8JFD4yt7Ic
        7GUWPpqOUHcdChk+ovpp3wz+Gz4yS/q6IUsfPpxwTavvfSE+NSoyMAOOIz4G6McqXdwlPtF0Ba1V
        cCg+3lqffyJSKz6v24BH8YouPnVkT2CCEjE+IkKdteoVMz581tIJG1Y1PgpZblQ+2jc+3lYD8leq
        Oj7koR8pXc89Pk4pzdioqUA+oLeTzLOgQj5p0mB1EdNEPj/kOsXAR0c+HOY/4pMGSj7Ud9AOSRhN
        PmXE1EBTQ1A+vJRMw0wuUj5GFHilLFNUPrVvQeLGuFY+bp6Ol71mWT7TLHVVmWVcPs2XUkvkvl8+
        UZyqVKS+YT6FSptLWdZjPiZsKBY7LWY+9lL98LzKaD7LeasDM7drPhXN2+bs+24+aOOgpalRcT6d
        fd+PhFxzPv9IJFAIpXU+j57JYXoyeD6TlLrF+wx7PoM8vuWiPX4+qmynQkzngD612hMOnOWCPkBe
        yQAaIIU+p8zS7d6dhz4Gxpnp2WaKPvxaNI7pg40+kGQ/HXx/kD489vrSjXGSPn1e8RZcnpQ+dZch
        JtQMlz7rZw9btMSZPqjh49akzpw+dZ6GiSkaoD6JG5VZSACiPjzYs/y6H6Q+1jWFJUR/pj6Fteef
        ciapPmxTomK5Haw+UOizd4purz4kQXuIupGxPgJScZQjpLM+FjVFjRn1tT7f3ELU/Iu4PiiAU3wM
        cbs+9VkMjYCtvj7HOkqv0yXBPkSP8DWDK8M+V5/ngT9uxT4PwfmmO/XHPi9n4RKEyMo+wCYOLxjx
        zT60xR2Eg7zQPtONjavHtdI+afIKqKHq1D6N4hhWGGLXPufbTLUGJNo+m8Ko7DQ53T5jDhshulXg
        PmzPeS/fQuI+bmxTISxq5D4r5nCrfNLmPihV1o57g+k+3114A7uF7D6PnxQE0OLvPn2FDWm40vE+
        BDhricvs8z6LNDz5Ukb2PltTPmPK5vg+JiGVW4/W+z6WX/4D/B7/Pqs2KWpCZQE/xgQV82xyAz8T
        MtkWhr0FP3XLHIvbTQg/3yR8g5crCz+f/R6y2l8OP415p6xs+hA/7JhQ5f36Ej+Yj5hdATgVPzsK
        T/CXuBc/h4gRrLmEGj9IAjA0T6UdP55h3g8nkiA/lPCQWGyGIj8VOZ+lsLUkP6OFewrpJic/rQy7
        pNzhKT+T2R1hPe8sP6Y9MNZhLDA/KXwDtKYUMj+uadtCgDY0P/sUq9u4mDY/I5uS1+dCOT9bjse8
        iT08PzuVVkUbkj8/bhboypulQT+KbwwCXbpDPxUO9+zxDUY/mymwRcOnSD/FoNhzGZBLP1z7b+02
        0E4/6kr52To5UT9BrNwlNEFTP+nES0t/hlU/bWeKg1cQWD87WbxX0uZaP8cZq1/5El4/qojkhHPP
        YD8cYw1k88piPzXyPoRMAmU/mKpttY18Zz/r+KnamkFqPya9yApGWm0/Ht/R0zVocD8q5rPiiFdy
        Px6G+aJFgXQ/y5IIjE/sdj/pMMkLWqB5P1gU/AwBpnw/yOb6MHIDgD+quIc14+aBP81vNC1XA4Q/
        o94NQYdfhj8tSm6T9wKJP2UutS8P9os/AO6gzDJCjz9CPkFb8XiRPxLlRyBuiJM/4e7qkx/WlT8G
        bW2vW2mYP2BRheNVSps/Mp5dNjmCnj9okQi7og2hP8C5TO53EKM/KXqSxgNQpT8WeoQvb9OnP0aN
        HDy7oqo/T7M/pNrGrT9tHfQh56SwPw5WT3tim7I/pfRamh/NtD+f6dpxG0G3P3ftX+wl/7k/y2ZW
        zfoPvT9onJbArj7APxzhkxocKcI/wzLwTF9NxD94K5dfSrLGP2Szl0J9X8k/dkJnFn5dzD8PMjhR
        1LXPP24164uTudE/ZNBXla/Q0z91AYlp5ibWPxsGtSSpw9g/FUXDjUmv2z9jMuuUFPPeP2I4GPm3
        TOE/l+oHof1W4z/+VOiE2p7lP9+GrgySK+g/V6A150IF6z/XLSbmADXuP5QvRfN44vA/i7kPETfg
        8j+ACSgoEhr1P4c98wQhl/c/bnALeFBf+j+j31aTfHv9PxWziHDGegBAQp5R90lsAkAKUdxHeZgE
        QA5V86Q/BgdA0dUzM1m9CUDC6iOba8YMQBPbecmQFRBAWDjO0yT7EUDQCrRT/BkUQC4gvg3YeBZA
        ltp3pUQfGUAIlDKoshUcQAWZpm2RZR9AaBsAkraMIUADuIQziJ4jQPnktObU7iVAkpHJ8fqEKEBR
        dAYNN2krQA6SSZ6+pC5Ae71Hhu4gMUD2kmhECiYzQCrvUVohaDVA5N+pzWTuN0CxgPq/3sA6QBHS
        JQmM6D1AHTxna7y3QEChWu5VcLBCQOBsAxOp5ERAW2ekfWtbR0D+zlJXkBxKQBBSSUXdME1AJJUN
        YBBRUEAuZVqnqD1SQJqbGThYZFRArQfh0fjLVkD+fWYFM3xZQNrpPJiWfVxAKuvhyLXZX0CTj/jk
        oc1hQFnQx2ob52NAf3PKIvc/ZkBCLeGUrt9oQL++1PGczmtAwYLvrxkWb0Cwon4lS2BxQBrlN8Pf
        bHNAJVfJTVG3dUBCcxtl60Z4QApmGujVI3tAhfPf7S5XfkB7yH7nk/WAQC+cr82S9YJACJMTsvIx
        hUAzxolm0rGHQEwfULMnfYpA/FOpsNicjUCdr+kObI2QQKWKx4cigZJApg2PLcevlECLS0EXTSCX
        QL+LDCp52plA0gdC1/rmnEDK+5/iwyegQKMds119D6JA56bHGbswpEBGCZJ/RZKmQMFLbr2xO6lA
        L7Nh7Xk1rEDASyUUGImvQP9SmSeSoLFA89P3SLu0s0AM9bUupge2QGHxZnW5oLhAyj9cJzuIu0B9
        3+MVa8e+QNC8/SZQNMFAb3QjA7U7w0ANYZQ3WoDFQGO2HO14CchAvlYWXiTfykCRJNWNYwrOQBpt
        OQSnytBALG5EA5bF0kCjSZktTfzUQKprYk/ZdddAvrMSCxw62kB+nOHy5FHdQIBnA8yGY+BAYKWI
        dExS4kCDCaAha3vkQM4YRVPE5eZAAbWXRAmZ6UBPFTJr053sQDVzENq//e9AROOg78bh8UBy/fCe
        oP3zQOHGrjgkWfZAqZbsudP7+EBJW/vHE+77QAi6IGtGOf9An0UgePRzAUFLo1Go2oIDQUT0HcXj
        zwVBusatr2NiCEF6u2OBi0ILQWoQaqKDeQ5BwdDrecQIEUHXwSa1BgsTQcEycUDuSRVBG8Q3/KHM
        F0FlvIGyIJsaQSQrRo1avh1B68C5xiagIEFPLaiuEpYiQQN0xnEvxyRBI/4nBHg6J0H0bHMVuvcp
        QV5U1+quBy1Bgjqgkws6MEHYuCXt7CMyQZSLbZyTRzRBkS3zts+rNkFjto3/Plg5QdOPxCdlVTxB
        L/Vl7casP0Eo7lw1hLRBQZFw7XwHy0NBF6OQi5MgRkE8HKNdl7xIQXXvD1pip0tBXVdfyT7qTkE9
        I9+1x0dRQeXKG0Z4UVNBsAo6fa6YVUG5WmGwqyRYQW57Bj2M/VpBoAlxXWEsXkHIiYcEp91gQVBe
        Rp7T2mJBLCY/CAwUZUEQV8UIZZBnQUoLSS3JV2pBTrY5ABJzbUE+1gAcEnZwQfnhbZwHZ3JBTgTt
        JpiSdEGr2KQErf92QQd77SQAtnlBTaFeuDS+fEG6HlpZ+RCAQSvckcUC9oFB+zyITj8UhEFnhU3L
        bXKGQc2kt7cYGIlBkZZSOK4NjEGNKFXzmlyPQbsYDQq0h5FBnbxZbO6Yk0FVnjgKkuiVQaqNaQ/7
        fZhBp8o22mNhm0Gqdocv/5ueQThUAsMKHKFBaa/N4pIgo0GR/dLxBGKlQdU2KuiP56dBpQjVmzu5
        qkGmlONTAuCtQWO42K/2srBBohmkhhqrskGx2Fgyst60QVqIAo3AVLdBG5ayGhwVukH/rpP/hyi9
        QdbIx/NnTMBBBbMynHM4wkGzzsT4hV7EQYzNb9R2xcZBmzM7kOx0yUGkAguAdHXMQe3B5Cae0M9B
        Npq31IzI0UGzydHrbOHTQbQ+Cx2dOdZBGaYDzpTY2EG6e9fMrMbbQXfNIOU5Dd9B+ni8S1Vb4UGi
        Qw8pVGfjQbcWRkoeseVBiTgjOv0/6EG9BI2DFhzrQc+vTaOFTu5BOLSJhLzw8EFUeQdCKfDyQY62
        OcHlK/VBtqijyw6r90EG48njl3X6QTGzm5dklP1BQkWpZ7KIAEKMIHc52nsCQn1bi2XfqQRCnfUG
        B7MZB0K8ilPLF9MJQofoCKm63gxCKdx4QCcjEEJoNJaAVQoSQtjuYpb3KhRC5Iji+tOLFkJXUkuy
        fTQZQq1OI2tsLRxCj9qVdReAH0Jab3H0iZshQrJ8dCsbryNCWDiPPFwBJkJuc3qnsZkoQhEJ5Rlf
        gCtCwJwswaG+LkJQDFTbZi8xQkPbG3I3NjNCVp/t5DZ6NUKbybRMnAI4QmsAqpV41zpCuBNoL9AB
        PkINa0Hi28VAQhoWiio6wEJCcVM+jU/2RELcw1HTJm9HQldVPl+fMkpCGyJWP4ZJTUIYNn4a2V5Q
        QtcuBIURTVJCO3kNTJJ1VEISALv4Ot9WQt8KBZS6kVlC7e4SH6iVXEIpVlHunfRfQirJMh+s3GFC
        AEUxsuv3Y0KZCxADw1JmQnNUNuqx9GhC1suVpxrma0ISUb+VXDBvQj5aggH5bnFCh+/ax0h9c0IW
        x969qcl1Qir2NK1tW3hCi+OXWMM6e0KxTPCK0HB+Qgt2k5znA4FCSrC5CZcFg0Lu7u922kOFQocp
        +rnWxYdC5BWUVIiTikIjgYHj3LWNQq/XusZnm5BCWE0vZsSQkkKSSyf7QMGUQpd8l3vWM5dCfGXf
        XFDwmUKe+qVmZf+cQhnEkLlpNaBCIdWVOr8eokK3E3eTyUGkQkkkzudWpaZCLmDZzQJRqULB9uOI
        Tk2sQp7RHh+8o69CrBiWUHavsULbBuYBYcWzQmg+u3tCGrZCMu8zm4e1uEJXaexnfZ+7QoK1f4Nr
        4b5CaH+O29hCwULnz6d+9EvDQraCmDiEksVCEP9RTMcdyEL9C4zG1/XKQuW26krEI85CItAJdtbY
        0EKOPke1cdXSQo7kkKAHDtVCL3O8+KqJ10IkWrUIRFDaQhYYLdSpat1CKY1FH19x4EIK6uHBxmHi
        QoetqLO4jORCBtysRBz55kK04KIvqa7pQoZ16C0AtuxC9YPHOGMM8EJC0nQu4vDxQoaVuOyDDvRC
        TGqtXQVs9kLRSxHW7hD5QtzoWhKsBfxC0GUECKdT/0IGlznwsoIBQ6dmez5WkwND9ZxN91DiBUMJ
        oZAs/XYIQ10VQeOSWQtDmlQPQEKTDkN13xNlKBcRQ9a5rRAdGxNDbynrR+pbFUOIF+z1vOAXQ/F8
        0aWasRpDXR2VDnvXHUMvjg5RMq4gQy1dPz3GpSJD4aSOBb3YJEN3A6iDF04nQ7WC0P6pDSpDz8/Z
        GjUgLUNEYejbwEcwQ533lQ1AMzJDKHHbYrVYNEPpTJWy9r42Q5WDvC6obTlDuco4ulRtPENIPl8d
        icc/QwCA4Dd5w0FD
    integration grid lopez: !!python/object/apply:numpy.core.multiarray._reconstruct
      args:
      - *id001
      - !!python/tuple
        - 0
      - !!binary |
        Yg==
      state: !!python/tuple
      - 1
      - !!python/tuple
        - 151
      - *id002
      - false
      - !!binary |
        ldYm6AsuET5rpdOAHNUYPrYc2Chp8iE+CW4qrvHwKT4dkU7Wir8yPhCJ1glzGTs+2DwaJtWVQz4o
        94VsMU9MPu/vLMewdVQ+Klx7JsSSXT7K3P0Ui19lPvgfeUnJ5G4+NLf1TNZTdj5tuPz68iKAPhZw
        PcYJU4c+VHDsVWPbkD530Qosol2YPiOCTc8PnKE+HHWHuiF0qT4KILmGVmWyPqvWb34Ql7o+Y+Ga
        z5k3wz7llImX/MbLPjJNOWFAE9Q+CFsRfnoE3T7xaOOItfjkPm1bQkslUO4+DM9bXmno9T5SyxoF
        n6r/PhfomvrQ4gY/HjHBdkiKED84BwOxZugXPxrEfOpVRyE/41oiS6r5KD8Ly3w41AwyPxTkH0ch
        Fzo/wy0Y2yPbQj8j8vEYV0FLPyB/YZupslM/Uv99bt14XD8eH2PCzpNkP7I0w3ZMvm0/Tx2OTAF/
        dT91WjIsQxJ/P5z7dB+0dIY/xsIr0bM6kD8vkOxBX3WXPyu5fasz9KA/vHKuFoCBqD/QF3LD+7Wx
        P5qZmZmZmbk/TUAvw2qAwj8+Eq+fNL7KP5ma3Y7jU9M/YBjqGeDv2z+B1PZ0zTDkP2EyFFsxL+0/
        bk29YJQX9T/hb7dgxHz+P85V9A6pCAZA2G1SHzzZD0DdA/NEgQQXQAqs+WihoiBAr4MsCpgLKEAW
        Z7AmxWAxQHz/SeRtHjlA6qR8K2YnQkDWevMViT1KQMG7uX/l9lJANE1m4HVpW0BO69WAqM9jQPFF
        +cfGomxAfPjWEhmydEAO86/bFOp9QBHGFNSlnoVAAAAAAABAj0D+LUNTwpWWQOKadp6XUqBAN4Pl
        R+eXp0BhIdGHKA2xQGhOR82SpbhAwBPX3w3QwUAFFRegSL/JQB/2B9ymm9JAvkLBXpLl2kCLDx7y
        VnDjQDemqsz/GOxAgsakCIZO9EC/VmsYJ1r9QNZVuqmgNgVBX1QsJaWpDkFPpxU4GCkWQbnbpesO
        BCBBsTlqJmMmJ0GfcDYyHrswQb0hRTH9LjhB8OSN01l6QUFUan+bZ0NJQRokFjsfQlJByFtjbSlk
        WkEAAAAA0BJjQaWBo7bPkWtBEcjvFNLsc0GtqkPT7cx8QT7JyvmP0IRBNouVsx0WjkGbC77veL6V
        QSgtbCcAbp9B8LeXLwG3pkEQdVWWnmqwQRWyjSKiurdBByirIEImwUFuJI2d2snIQceDp1xG6tFB
        5vUvHy/l2UGVTvALC7fiQZ1udhAqDetBofanNvSM80FMq2QIXEL8QWN0S1xqbARCjeOkElyFDUL3
        tb6m2lUVQksJUfnH1h5CEYegH7dJJkLF5ANJohswQgAAAOh2SDdC4Us5B7/TQEK5s45yllJIQuwM
        MigUlFFCqQpswJdoWUI4Btyg/1xiQit9p6ACi2pCkMYPmOMuc0IJtJjyZLp7QmVRg5YmCoRC4vct
        61L3jEJc7uC4M++UQvnKUl1nQp5CXHwa5HrepUKrC5IFRJyvQu3XEvxw2LZCA0mM7MiCwEKQCLgc
        kN3HQo1xH6yAP9FCkhuq1Vfu2ELjI2FypQTiQiDNi2hNC+pC7LXrjZfS8kLgdBsK/DT7QsJAIZq7
        qQNDAAA0JvVrDEM=
    scaling correction: 6.283185307179586
  atmospheric showers:
    atmosphere: !!python/tuple
    - CORSIKA
    - !!python/tuple
      - Karlsruhe
      - null
    interaction model: SIBYLL2.3c
    native mceq: true
    particles of interest:
    - numu
    - nue
    - nutau
    path to mceq: /home/kruteesh/miniconda3/envs/pdm/python3.10/site-packages/site-packages/MCEq/
    primary model: H4a
    theta angles: !!python/object/apply:builtins.range
    - 0
    - 91
    - 1
  general:
    channel: All
    config location: ../run/config.txt
    debug level: 40
    density: NFW
    detector: POne
    log file handler: ../run/pdm.log
    pone type: old
    random state seed: 1337
    sigma_model: prada
    year:
    - 0
    - 1
  pone:
    aeff location: ../data/
    high E sigma:
      nue: 0.07
      numu: 0.15
      nutau: 0.07
    low E sigma:
      nue: 0.25
      numu: 0.25
      nutau: 0.25
    smearing: smeared
    specific particle scaling:
      nue: 1.0
      numu: 1.0
      nutau: 1.0
  pone_christian:
    aeff location: ../data/
    angles:
    - 0
    - 1.46928358
    - 4.41172579
    - 7.36588583
    - 10.33989089
    - 13.3423638
    - 16.38266985
    - 19.47122063
    - 22.61986495
    - 25.84241287
    - 29.15536543
    - 32.57897039
    - 36.13881466
    - 39.86834155
    - 43.81306146
    - 48.03811117
    - 52.64314803
    - 57.7957725
    - 63.82304783
    - 71.5713304
    - 90
    hit: []
    module: []
    pos res: []
    spacing: []
  runtime:
    random state: !!python/object/apply:numpy.random._pickle.__randomstate_ctor
      args:
      - MT19937
      state:
        bit_generator: MT19937
        gauss: 0.0
        has_gauss: 0
        state:
          key: !!python/object/apply:numpy.core.multiarray._reconstruct
            args:
            - *id001
            - !!python/tuple
              - 0
            - !!binary |
              Yg==
            state: !!python/tuple
            - 1
            - !!python/tuple
              - 624
            - !!python/object/apply:numpy.dtype
              args:
              - u4
              - false
              - true
              state: !!python/tuple
              - 3
              - <
              - null
              - null
              - null
              - -1
              - -1
              - 0
            - false
            - !!binary |
              OQUAAH6QXDO4b0royidGbRtWhmJH4skVCUWvxvmWXkrgR+S2M062oV8SHCyGFoEB6pil0voEiTSw
              wI6oSUhPQngM7vCYv/Wt1AG0Why20gYg1Vo/tTXE7uSWIXRwFUSHEnh0e5iKulV3j/P836wIABeL
              CHbLpWc5NQwrSqOk75GlHA0/OpoDSWlsktf1f68XzZhqHwb+bhqEbuVNoMdMpRJ1jhtD0ljnajNP
              jjPwmc8cc5UaOWZUbUZMiLgDe3KElRAFTpW+6mGWZKbn/GuMDKB2FF269vXWM1giO1YXJaNcKGpK
              6rEGgsJYMZ/7wzakeZJKgMSdLGv3q2uV6PQzopLa9jbbXVpcRLHWVHzddFKVRw7dc4QLGKXMkcnF
              k5QtAboVpXj+MhyinZT/0FlkmSbSZkWwyBibiHAMCfct2a35PvptKZA1qEnkdRwgIq1bWSB5rTx3
              l4nMOUTCApaDECJBP+FeW7Zr1eNwqS7zWDKCj32F3R3VyPSWIsLSvjBvnzQfMBza0TNuCUiG/4Nq
              WZoBmwsnDDmL/5cEZp3lGAVZnP7Y69mVHCG9KpI6Wgl6syyXgkJbZlHd3A2nanvEcaf0fYyK0iAi
              5OpkAWewwCoXuD1DI2d33xbQK2KKaEaSIAZwn+OdaF2kPPJ/lDqJYUXakGFRgdCqPXG9hlplETcC
              J5DG5uzF+NsDICjquK51O7hRIcxCcs4xI37AQaX3gPcO+jH8Fmsm9u1waP4RqFknkelii5rdBGVc
              72AD+JwNv3S8a4e9BbUL81LIuyoBcvZlaT2j4N8z5dv/LvBO4hZIlReMy4A4MbFzxNHV50kmpXQM
              2xvc4E7gwDZxZIL5KRT289a0ZpPL1XUgNqx6sM6vDhEssHJfvuCC52psDIiSDsZVXZQrPnhz7+zC
              22etf66j8/ruKHLaL/trYdW3R39U+3cP1Rwin4VwxYf2pM5BJ0WQQ7Oe9sgmzAjbUFi7UK0xZLLE
              QecvDtfKJEFXf9rGv72UEY3YKHPBZAge3msJlrA+LO/xVR0NW+oC5OGKioI2+TAPFAwGsfHouWYn
              JSIGIoh2jCwcgNCS176lT11EgDDrRbIsRjII5Xi0wbeKYZA/UL5rN6xcpJd/9bY0EL3YH4Wk1zuH
              /XoPK3he76LEiTXsjbCDpYEUOKVrJF7FR3XyQh8GRl0mPNdvf/CQ+r9asgrk/v4YHhiJRGHAeBdZ
              Ce4obw0ywKTRxwaNJsJhvxzdclFawnDiB0/tiuTaDBTgYHL2fLUu2wmTBwN81OsQ3DElAb1oX4hN
              i+tN76GNqnW4A3+52lkV80yPiQxVTbx+DWnoSjgVBCzQhLkhwRBenFLW27iwHsTFzcyFgq9d+OVH
              aVj1YT5gRjq0+DzrLEYVdUfks/hzyXZOp5fLCCpzupLPY9D3h11/n/qcXNKlJlg7g42Plf+mr3a5
              PBp7eEi3BSfTFZVeSPaFpK5XbkSMnQVe1mvGa7ODeZIvxgxmvmzuc8TkeV/lPbg4YMfMnKrVN9Xu
              i47aI/WgT6WSsU9wH7t91C3PgAzj0uuD6JwL8Nus+1kn3cHhWoTQDKlijjvVXpiw/Hq/gmB1T5PV
              wtgxNdyb2nvI0HVQY/4pv/+yK4setZo+XouY3koOqP9Ft8+geaZBuiYsPZjSOkqU+BcSnEeIM8W9
              WgjzTL4nGmjnI379t9GO6Kne3zVXnnCGRtmfl+unMtdgtTKXrYQzV7WgmaCLgWxRRp2s/1NQiHWy
              nro6FYzho9TbdJvGWXTzv1B1qzeZQFb6IY5IrLuLjJ9Jwk3ufUAK95pcX8J3BGZDHOhkS7KnCyYr
              OfnPJ9TV+90rbqzmId8SLF+ywYIMJWpbfZQPXkny/9GQMqqq+RY0ZDjJn6lDb1BnbDgUsckiL6J7
              XsTNvX5PepKdU5M3Oz/u7DANZOoj3tpX3TqC9ND+yt+hHV8DrGJ3OPAIdOxIpC2MEkt7EshmzycJ
              FXlx8ztPpPz9Pyly5EM+c1AJ7aYbCfmxuPAbA/MlIa5GfxvU8Uy+maCKqUTGJyuJlvhHhuJCHnEk
              0fImtFdhapCXkvWdwbl8s87nwKV7RDLqyorWPrcwbVEqecXegRG+774KKtPXGWgNGmp0rtj7H/9T
              MWmaA+W6jZq0XUFj+9biD5olaG67ynO2kv06imYb6Mlw3f46yE4e+bC6rExvYuV2AbUuw2Z9Wct2
              hywoLJlJKvv7NdJ4IuxWXlzrpu6tzCWJ/lace9FX1v3dBurchHQ7cygWQqL4gCGTy15hRHRVTuTN
              0uzP3KxU81z/214cDD/FgIfJzsXJE/dJGBolX+uEFmvZdMeQtxU4n1e3mN1dDK/GxktegoEcz75u
              xjZ0YNWkN7FW1UI1LMZcmqESf+eWD7PEpeGaxwuGhcmf1lKROTE2hAcZnIDfB4L9PYkNy6GevCiC
              VOdFH+05rIIVxHwiFJu7mHr3nC3v7rM7GS1kNawtV3QRnJHJ6zZ2TyTmbNU2rjI6IqOzoHV9C3+a
              lNejz/m6MoNXHTyIo/0Tgk5NeIoWGT9OwLsno54eUchJZxLHJbTtNMyk6JQCz8N11i/PccTe/N6E
              InLgxNI2Ro6jTuoinGWfi7a92hsHlyKWnEa6+BSHhJpo8TA4qGZDZse7lCIrA1TakbffK726I5Um
              WF0K9tgKZzA15wKD7OopeLB/si3z5MjvlrdaZ2XIMUyUrcA63aIkDWZMsT8Q+qAWaAbY8HfD+9ER
              8LDojXxTHXJU16q78vz9orU8IWUKSYVIXrb5k1QwoNBco3IPVrEquS/sCTCXV5iM1kwMQOFh2qee
              G8CInGLVUwLvQDPcX5RZRBnimbJwpO7qMAboA//8fOGvVoWn4YiSMlVF2G/XSG6B3gjKZl0hYsD6
              9pOowvA/Z7bARF6ObkacBGjs3evbI1RVCIYSs0702sBXkl6vuhg9gBIdOT+YnHCHhOCJ/eq4Am0y
              SVufUBwUlJMQEiIEAUkVWyP9eFISAiM0aGFtYzx4Emryux8azc39T2SyT/Vi1j5UBKZSt8pS6hD0
              L81I7JeAuVpqAxrdqov1ioYDNey4km/EB/mnHAQBu/UFKYUBPN6fF/DLpENacb8gyOSmcJTWxs/b
              emr0gRGo5ZR8O9/e+EUE4v/bl63V2iSP5Gp6VSxeqKMPQ73WVTnoW9xggnCSkdizVPPdxZsiWqpa
              XwdpwckixYH5R60YPiibU2/EU1XQfLaNGKlhPnGqbZrFUfu6jcs4fsHkqGKBJKrWbzd1raZPKIXB
              TZJ8uy0ASqBgzL3eWEmZDvCbFtHuYkS927Dl6sULbX6W+td5z6t4dQPmIJmr
          pos: 624
  simulation parameters:
    DM type k: 2
    high energy cutoff: 5000000.0
    low energy cutoff: 1000.0
    mass grid: !!python/object/apply:numpy.core.multiarray._reconstruct
      args:
      - *id001
      - !!python/tuple
        - 0
      - !!binary |
        Yg==
      state: !!python/tuple
      - 1
      - !!python/tuple
        - 9

pone_dm/config.py
# -*- coding: utf-8 -*-
# Name: config.py
# Authors: Stephan Meighen-Berger
# Config file for the pdm package

import logging
import numpy as np
from typing import Dict, Any
import yaml

_baseconfig: Dict[str, Any]

_baseconfig = {
    ###########################################################################
    # General inputs
    ###########################################################################
    "general": {
        # Random state seed
        "random state seed": 1337,
        # Output level
        'debug level': logging.ERROR,
        # Location of logging file handler
        "log file handler": "../run/pdm.log",
        # Dump experiment config to this location
        "config location": "../run/config.txt",
        "detector": ["IceCube",   "POne", 'combined'],
        "pone type": ["new", "old"],
        "year": [0, 1], # 2, 3, 4, 5, 6, 7, 8, 9],#range(0, 10),
        'density': ['NFW', 'Burkert'],
        "channel": ["W", "\[Tau]", "b", "All", "\\[Nu]\\[Mu]", "\\[Nu]\\[Tau]",
                    "\\[Nu]e"],
        'sigma_model': ['lopez', 'prada', 'mixed']
    },
    "simulation parameters": {
        "mass grid": np.logspace(2, 6, 22),
        "sv grid": np.logspace(-26, -21, 22),
        'theta': range(0,91,1),
        "uptime": 10 * 365 * 24 * 60 * 60,
        "low energy cutoff": 1e3,  # GeV
        "high energy cutoff": 5e6,  # GeV
        "DM type k": 2
    },
    ###########################################################################
    # Atmospheric showers input
    ###########################################################################
    "atmospheric showers": {
        # native mceq or built one
        'native mceq': True,
        # Path to the built version
        'path to mceq': '/home/kruteesh/miniconda3/envs/pdm/python3.10/site-packages' +
        '/site-packages/MCEq/',
        # The atmosphere
        'atmosphere': ('CORSIKA', ("Karlsruhe", None)),
        # The interaction model
        'interaction model': 'SIBYLL2.3c',
        # Primary model
        'primary model': 'H4a',
        # Angles of interest currently not custom
        'theta angles': range(0, 91, 1),
        # Particles of interest
        'particles of interest': ['numu', "nue", "nutau"]
    },
    ###########################################################################
    # P-ONE
    ###########################################################################
    "pone": {
            'aeff location': '../data/',
            "specific particle scaling": {
                "numu": 1.,
                "nue": 1.,
                "nutau": 1.},
            'smearing': ['smeared', 'unsmeared'],
            'low E sigma': {
                'numu': [0.45, 0.35, 0.25, 0.15],
                'nue': [0.55, 0.45, 0.35, 0.25],
                'nutau': [0.55, 0.45, 0.35, 0.25]},
            'high E sigma': {
                'numu': [0.25, 0.15, 0.10, 0.09],
                'nue': [0.15, 0.12, 0.09, 0.07],
                'nutau': [0.15, 0.12, 0.09, 0.07]
                }
            # Entries: numu, nue, nutau
    },
    'pone_christian': {
        'aeff location': '../data/',
        'angles': [0, 1.46928358,  4.41172579,  7.36588583, 10.33989089,
                   13.3423638, 16.38266985, 19.47122063, 22.61986495,
                   25.84241287, 29.15536543, 32.57897039, 36.13881466,
                   39.86834155, 43.81306146, 48.03811117,
                   52.64314803, 57.7957725, 63.82304783, 71.5713304, 90],
        'spacing': [],
        'hit': [],
        'module': [],
        'pos res': [],


    },
    ###########################################################################
    # Advanced
    ###########################################################################
    "advanced": {
        "integration grid lopez": np.logspace(-9, 19, 151),
        "construction grid _d": np.logspace(-13, 16, 600),
        "_d storage": "../data/",
        "atmospheric storage": "../data/",
        "scaling correction": 2*np.pi
    }
}


class ConfigClass(dict):
    """ The configuration class. This is used
    by the package for all parameter settings. If something goes wrong
    its usually here.

    Parameters
    ----------
    config : dic
        The config dictionary

    Returns
    -------
    None
    """

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

    # TODO: Update this
    def from_yaml(self, yaml_file: str) -> None:
        """ Update config with yaml file

        Parameters
        ----------
        yaml_file : str
            path to yaml file

        Returns
        -------
        None
        """
        yaml_config = yaml.load(open(yaml_file), Loader=yaml.SafeLoader)
        self.update(yaml_config)

    # TODO: Update this
    def from_dict(self, user_dict: Dict[Any, Any]) -> None:
        """ Creates a config from dictionary

        Parameters
        ----------
        user_dict : dic
            The user dictionary

        Returns
        -------
        None
        """
        self.update(user_dict)


config = ConfigClass(_baseconfig)
